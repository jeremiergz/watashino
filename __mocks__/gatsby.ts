import Gatsby from 'gatsby';

module.exports = {
  ...Gatsby,
  graphql: jest.fn(),
  navigate: jest.fn(),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
};
