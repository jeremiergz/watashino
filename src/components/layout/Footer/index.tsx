import Anchor from '@/components/common/Anchor';
import useSiteMetadata from '@/components/hooks/data/useSiteMetadata';
import clsx from 'clsx';
import React from 'react';

function Footer({ className, ...rest }: FooterProps): JSX.Element {
  const { repositoryUrl, version } = useSiteMetadata();

  const sourceCodeUrl = `${repositoryUrl}/tree/${version}`;

  return (
    <footer className={clsx(className, 'flex flex-col items-center justify-center')} {...rest}>
      <Anchor
        aria-label={`Go to ${sourceCodeUrl}`}
        className="border-b border-dotted text-2xs text-gray-400"
        external
        href={sourceCodeUrl}
      >
        <div className="text-sm text-gray-300">v{version}</div>
        source code
      </Anchor>
    </footer>
  );
}

export type FooterProps = React.ComponentProps<'footer'>;
export default Footer;
