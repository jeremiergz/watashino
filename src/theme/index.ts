const breakpoints = ['320px', '375px', '425px', '768px', '1024px', '1200px', '1440px', '2560px'] as const;
Object.defineProperties(breakpoints, {
  mobileS: { value: breakpoints[0] },
  mobileM: { value: breakpoints[1] },
  mobileL: { value: breakpoints[2] },
  tablet: { value: breakpoints[3] },
  laptopS: { value: breakpoints[4] },
  laptopM: { value: breakpoints[5] },
  laptopL: { value: breakpoints[6] },
  desktop: { value: breakpoints[7] },
});

const colors = {
  black: '#212121',
  dark: '#424242',
  error: '#F44336',
  gray: '#757575',
  info: '#2196F3',
  lightgray: '#E0E0E0',
  primary: '#31859A',
  secondary: '#17365C',
  success: '#4CAF50',
  white: '#FAFAFA',
} as const;

const fonts = {
  main: "'Open Sans', sans-serif",
} as const;

const fontWeights = {
  bold: 800,
  regular: 400,
  'semi-bold': 600,
} as const;

const DarkTheme = {
  background: colors.black,
  breakpoints,
  colors: {
    ...colors,
    text: colors.white,
  },
  fonts,
  fontWeights,
  svg: colors.white,
  type: 'dark',
};

const LightTheme = {
  background: colors.white,
  breakpoints,
  colors: {
    ...colors,
    text: '#262626',
  },
  fonts,
  fontWeights,
  svg: '#262626',
  type: 'light',
};

type Theme = typeof DarkTheme | typeof LightTheme;
type ThemeProps = {
  background?: string;
  colors?: keyof Theme['colors'] | 'inherit';
  fontWeight?: keyof Theme['fontWeights'];
  svg?: string;
  type?: 'light' | 'dark';
};

export { colors, DarkTheme, LightTheme, Theme, ThemeProps };
