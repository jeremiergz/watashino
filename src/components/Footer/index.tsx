import Link from 'components/Link';
import Box from 'components/primitives/Box';
import { graphql, useStaticQuery } from 'gatsby';
import React from 'react';

const Footer: React.FC = () => {
  const {
    site: {
      siteMetadata: { license, repository, version },
    },
  } = useStaticQuery(graphql`
    query FooterData {
      site {
        siteMetadata {
          license
          repository {
            url
          }
          version
        }
      }
    }
  `);
  const repositoryURL = repository.url.replace(/git\+|\.git/g, '');
  const sourceCodeURL = `${repositoryURL}/tree/${version}`;
  return (
    <Box
      as="footer"
      bottom={0}
      left={0}
      padding={{ _: 16, tablet: 32 }}
      position="absolute"
      right={0}
      textAlign="center"
    >
      <Link borderBottom="1px dashed gray" color="gray" external to={sourceCodeURL}>
        {`v${version} - source code`}
      </Link>
      <Box color="gray" fontSize={14} marginTop={1}>{`licensed under the ${license} license`}</Box>
    </Box>
  );
};

Footer.displayName = 'Footer';

export default Footer;
