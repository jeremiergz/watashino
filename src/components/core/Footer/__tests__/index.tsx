import * as Gatsby from 'gatsby';
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Footer from '..';
import { DarkTheme } from '../../../../theme';

describe('components/core/Footer Test Suite', () => {
  beforeAll(() => {
    const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
    useStaticQuery.mockImplementation(() => ({
      site: {
        siteMetadata: {
          license: 'Test-Licence',
          repository: { url: 'git+https://test-repository.git' },
          version: '3.16.120',
        },
      },
    }));
  });

  it('matches snapshot', () => {
    const jsx = (
      <ThemeProvider theme={DarkTheme}>
        <Footer />
      </ThemeProvider>
    );
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
