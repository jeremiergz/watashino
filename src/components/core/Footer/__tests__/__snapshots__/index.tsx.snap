// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Footer Test Suite matches snapshot 1`] = `
<footer
  className="sc-AxjAm hiAgES"
>
  <a
    className="sc-AxjAm sc-AxirZ cZNOyj"
    href="https://test-repository/tree/3.16.120"
    onClick={[Function]}
    rel="noopener noreferrer"
    target="_blank"
  >
    v3.16.120 - source code
  </a>
  <div
    className="sc-AxjAm iIgCKo"
    color="gray"
    fontSize={14}
  >
    licensed under the Test-Licence license
  </div>
</footer>
`;
